### ISSUE TITLE
OpenAPI error when reader configuration property set in bootstrap.properties

### ISSUE THREAD

**Issue by jpfloresibm** [OpenLiberty/open-liberty#15765]
**Describe the bug**
When the configuration properties for the open api reader are provided in the bootstrap properties, the following error is observed:
```
[2/1/21 9:23:00:987 PST] 0000003f com.ibm.ws.microprofile.openapi.utils.OpenAPIUtils           E CWWKO1658E: Failed to load the OpenAPI model reader class: com.ibm.mdmx.openapi.extension.reader.IAMSecurityOpenAPIReader.
```

However if the configuration is provided in the `microprofile-config.properties`, the error is not generated.

**Steps to Reproduce**
Create a custom OpenApiReader implementation and define it in the bootstrap.properties file. Start up WLP. Observe messages.log for the following error: CWWKO1658E

**Expected behavior**
The CWWKO1658E error should not be generated when the config is provided in the bootstrap.properties.

**Diagnostic information:**
 - OpenLiberty Version: [20.0.0.10]
 - If it would be useful, upload the messages.log file found in `$WLP_OUTPUT_DIR/messages.log`
[messages.log](https://github.com/OpenLiberty/open-liberty/files/5907213/messages.log)
 - Java Version: [1.8.0_251]
 - server.xml configuration (WITHOUT sensitive information like passwords)
[server.xml](https://github.com/OpenLiberty/open-liberty/files/5907231/server.xml.txt)
 - bootstrap configuration
[bootstrap.properties](https://github.com/OpenLiberty/open-liberty/files/5907235/bootstrap.properties.txt)
- custom reader classes
[BearerTokenSecurityOpenAPIReader.java.txt](https://github.com/OpenLiberty/open-liberty/files/5907271/BearerTokenSecurityOpenAPIReader.java.txt)
[IAMSecurityOpenAPIReader.java.txt](https://github.com/OpenLiberty/open-liberty/files/5907272/IAMSecurityOpenAPIReader.java.txt)

**Additional context**

**Comment by Emily-Jiang** [comment]
@jpfloresibm From the files you attached, it seems that you are trying to use MP Open API. However, in your server.xml, you didn't list the feature mpOpenAPI-1.1. Please try again by adding
`<feature>mpOpenAPI-1.1</feature>` to your server.xml.

**Comment by jpfloresibm** [comment]
After trying the mpOpenAPI-1.1 feature the issue still persists. Note that this is only happening if the config is defined via bootstrap.properties (the class still loads and the extra markup is seen in the api explorer page but the error message is showing up). If defined in the microprofile-config.properties the error message is not generated.

**Comment by jpfloresibm** [comment]
Hi guys. Any updates on this?

**Comment by Emily-Jiang** [comment]
@arturdzm can you investigate this and provide a update?

**Comment by jpfloresibm** [comment]
@arthurdm @Emily-Jiang are there any updates to this bug? Are service logs keep generating these errors and would be nice to put in a fix for this issue.

**Comment by arthurdm** [comment]
@arturdzm - fyi ^

**Comment by Azquelt** [comment]
I think there are two things happening here:
* when defining config in `bootstrap.properties`, it will apply to all applications
* `openapi-3.1` is trying to process some internal web modules, as well as deployed user applications

`openapi-3.1` reads the config from `bootstrap.properties` and tries to load the configured `OpenAPIReader` from the internal web module, which does not have a class by that name.

@arturdzm I think we may just need to add additional context routes to the ignore list in [`OpenAPIModuleListener`](https://github.com/OpenLiberty/open-liberty/blob/integration/dev/com.ibm.ws.openapi.3.1/src/com/ibm/ws/openapi31/OpenAPIModuleListener.java#L69-L78). Does that seem right?

**Comment by jpfloresibm** [comment]
hi @Azquelt has there been any progress on this bug? We still keep seeing it on version 21.0.0.10

**Comment by Azquelt** [comment]
No, @arturdzm is the person who owns it, I'll follow up with him.

**Comment by InhexSTER** [comment]
@Azquelt I agree.

the `bootstrap.properties` is used for all applications, so configuration is applied to all WebModules.
Each application can have it's own reader, it's own fitler, exclude package, etc. So, logically these settings should only be inside each application's `microprofile-config.properties` and not in bootstrap or system property level.

The list of properties here is really meant as application setting https://download.eclipse.org/microprofile/microprofile-open-api-1.1.2/microprofile-openapi-spec.html#_core_configurations

There are few properties that are global for `openapi-3.1`  and could be in bootstrap.properties feature such as :
mp.openapi.extensions.liberty.public.url=myAPI
